dataplane/src/drivers/kernel.rs

```diff
@@ -42,20 +48,21 @@ pub struct Kif {
     sock: RawPacketStream, /* packet socket */
     raw_fd: RawFd,         /* raw desc of packet socket */
 }
+
 impl Kif {
     /// Create a kernel interface entry. Each interface gets a [`Token`] assigned
     /// and a packet socket opened, which gets registered in a poller to detect
     /// activity.
-    fn new(ifindex: u32, name: &str, token: Token) -> Option<Self> {
-        let Ok(mut sock) = RawPacketStream::new() else {
-            error!("Failed to open raw sock for interface {name}");
-            return None;
-        };
-        sock.set_non_blocking();
-        if let Err(e) = sock.bind(name) {
-            error!("Failed to bind to interface '{name}'");
-            return None;
-        }
+    fn new(ifindex: u32, name: &str, token: Token) -> io::Result<Self> {
+        let mut sock = RawPacketStream::new().map_err(|e| {
+            error!("Failed to open raw sock for interface {name}: {e}");
+            e
+        })?;
+        let _ = sock.set_non_blocking();
```

Comment:
The let _ here is unecessary unless there is some lifetime issue I am not aware of.

Reply:
I thought it was must use thingy, but I may be wrong. Let me check and fix